corehq/apps/callcenter/utils.py

```diff
@@ -189,18 +178,23 @@ def valid_element_name(name):
     return fields
 
 
-def _user_case_changed(case, case_type, close, fields, owner_id):
+def _get_changed_fields(case, case_type, owner_id, fields):
+    changed_fields = {}
+    if case.name != fields['name']:
+        changed_fields['name'] = fields['name']
+
+    if case.type != case_type:
+        fields['case_type'] = case_type
+
+    if case.owner_id != owner_id:
+        fields['owner_id'] = owner_id
+
     props = case.dynamic_case_properties()
-    changed = close and not case.closed
-    changed = changed or case.type != case_type
-    changed = changed or case.name != fields['name']
-    changed = changed or case.owner_id != owner_id
-    if not changed:
-        for field, value in fields.items():
-            if field != 'name' and props.get(field) != value:
-                changed = True
-                break
-    return changed
+    for field, value in fields.items():
+        if field != 'name' and props.get(field) != value:
```

Comment:
I think `field != 'name'` could be removed if the next line is changed to use `changed_fields.setdefault(...)`.

Also, is it ever possible that a field that is not in `props` will be intentionally set to `None`? If yes, might want to change this to `if field in not props or props.get(field) != value:`

Reply:
no, I don't think that would be desirable